nurseconnect_resubscribe.py

```diff
@@ -0,0 +1,154 @@
+parser.add_argument('--is-url', required=True,
+                    help='The url for the Identity Store.')
+parser.add_argument('--is-token', required=True,
+                    help='The token for the Identity Store.')
+parser.add_argument('--new-messageset', type=int, required=True,
+                    help='The id of the new nurseconnect messageset.')
+parser.add_argument('--old-messageset', type=int, required=True,
+                    help='The id of the old nurseconnect messageset.')
+
+args = parser.parse_args()
+new_messageset = args.new_messageset
+old_messageset = args.old_messageset
+sbm_url = args.sbm_url
+sbm_token = args.sbm_token
+is_url = args.is_url
+is_token = args.is_token
+
+try:
+    messageset_schedule = get_messageset_schedule(sbm_url, sbm_token,
+                                                  new_messageset)
+except requests.HTTPError as e:
+    sys.exit("Problem retrieving the messageset: %s" % e.response.status_code)
+
+# get all nurseconnect subscriptions that have lapsed
+subscriptions = get_subs(sbm_url, sbm_token, {"messageset": old_messageset,
+                                              "completed": True})
+count = 0
+for sub in subscriptions:
+    try:
+        # check they aren't already subscribed to the new messageset
+        if sub_exists(sbm_url, sbm_token, {"identity": sub['identity'],
+                                           "messageset": new_messageset}):
+            sys.stdout.write("Subscription creation skipped - Identity: %s "
+                             "already subscribed to messageset %s\n" %
+                             (sub['identity'], new_messageset))
+            continue
+
+        # check they aren't still subscribed to the old messageset
+        if sub_exists(sbm_url, sbm_token, {"identity": sub['identity'],
+                                           "messageset": old_messageset,
+                                           "active": True}):
+            sys.stdout.write("Subscription creation skipped - Identity: %s "
+                             "still subscribed to old messageset %s\n" %
+                             (sub['identity'], old_messageset))
+            continue
+
+        # check they haven't opted out
+        identity = get_identity(is_url, is_token, sub['identity'])
+        msisdns = identity['details']['addresses']['msisdn']
+        if all(msisdn.get('optedout') for _, msisdn in msisdns.items()):
+            sys.stdout.write("Subscription creation skipped - Identity: %s "
+                             "optedout\n" % (sub['identity']))
+            continue
+    except requests.HTTPError as e:
+        sys.stdout.write("Problem retrieving identity or subscriptions - "
+                         "Subscription: %s Error code: %s\n" %
+                         (sub['id'], e.response.status_code))
+        continue
+
+    data = {
+        'identity': identity['identity'],
+        'lang': identity['language'],
```

Comment:
I just want to make sure that this is the correct key, I remember it being something like `identity['lang_code']`, but that could be a different project.